### PR TITLE
[EIP-1559] Step 7 - Check transaction format for send raw transaction

### LINKED ISSUE #691: [EIP-1559] Check transaction format for send raw transaction
### Description
We must check format of transactions received on the JSON RPC API.
We must reject legacy transactions if EIP-1559 is finalized.
We must reject EIP-1559 transactions if feature flag is not enabled or if current block is not post EIP-1559.

### PR DESCRIPTION
**Pull request by abdelhamidbakhta**
## PR description
We must check format of transactions received on the `EthSendRawTransaction`  JSON RPC API.
We must reject legacy transactions if EIP-1559 is finalized.
We must reject EIP-1559 transactions if feature flag is not enabled or if current block is not post EIP-1559.
- Updated `EIP1559` manager: added `isValid` method to check whether or not if a transaction is valid according EIP-1559 rules.
- Updated `EthSendRawTransaction` : call `EIP1559.isValid` against the decoded transaction if the EIP feature flag is enabled
## Fixed Issue(s)

fixes #691 

Signed-off-by: Abdelhamid Bakhta <abdelhamid.bakhta@consensys.net>

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/fees/EIP1559.java [hunk 1]
```diff
@@ -76,6 +79,23 @@ public long getForkBlock() {
     return initialForkBlknum;
   }
 
+  public boolean isValid(final Transaction transaction, final LongSupplier blockNumberSupplier) {
```
Comment (shemnon) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/fees/EIP1559.java:82]: This feels like it should be in `MainnetTransactionValidator` instead, with two new constructor args.  If we get a third format we can go to an enum set or bitfield.  As part of `validate(final Transaction transaction)` we check the valid types against the booleans or enum set/bitfield.  Perhaps we need to define two forks for testing, `EIP1559` and `EIP1559Only`

Comment (abdelhamidbakhta) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/fees/EIP1559.java]: Done

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthSendRawTransaction.java [hunk 2]
```diff
@@ -99,4 +114,26 @@ private Transaction decodeRawTransaction(final String hash)
       throw new InvalidJsonRpcRequestException("Invalid raw transaction hex", e);
     }
   }
+
+  private Transaction checkEIP1559(final Transaction transaction)
```
Comment (shemnon) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthSendRawTransaction.java:118]: I don't think we need any of this logic in the RPC stub. MainnetTransacitonValidator should be catching it when we try and `addLocalTransaction`.  The RPC stub should have no knowledge of the internals of the block chain.

Comment (abdelhamidbakhta) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthSendRawTransaction.java:118]: Ok done.

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthSendRawTransactionTest.java [hunk 3]
```diff
@@ -42,7 +43,7 @@
   private static final String VALID_TRANSACTION =
       "0xf86d0485174876e800830222e0945aae326516b4f8fe08074b7e972e40a713048d62880de0b6b3a7640000801ba05d4e7998757264daab67df2ce6f7e7a0ae36910778a406ca73898c9899a32b9ea0674700d5c3d1d27f2e6b4469957dfd1a1c49bf92383d80717afc84eb05695d5b";
   @Mock private TransactionPool transactionPool;
-
+  @Mock private BlockchainQueries blockchainQueries;
```
[the remark this answers is not in the document]
Comment (abdelhamidbakhta) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthSendRawTransactionTest.java]: Done

**Pull request merged**